app/src/main/java/ch/epfl/sweng/favors/database/storage/Storage.java

```diff
@@ -110,20 +125,74 @@ public String uploadImage(StorageReference storageReference, Context context, Ur
     }
 
     @Override
-    public void displayImage(ObservableField<String> pictureRef, ImageView imageView) {
-        if(pictureRef != null && pictureRef.get() != null){
-            StorageReference ref = FirebaseStorage.getInstance().getReference().child("images/"+ pictureRef.get());
+    public void displayImage(ObservableField<String> pictureRef, ImageView imageView, String category) {
+
+        if(Storage.checkStoragePath(category) && pictureRef != null && pictureRef.get() != null){
+            String path = category.toLowerCase()+"/";
+            StorageReference ref = FirebaseStorage.getInstance().getReference().child(path+ pictureRef.get());
             view = imageView;
             ref.getBytes(MAX_BYTE_SIZE).addOnSuccessListener(byteSuccessListener);
 
         }
     }
 
     @Override
-    public void deleteImageFromStorage(ObservableField<String> pictureRef) {
-        if(pictureRef != null && pictureRef.get() != null && !ExecutionMode.getInstance().isTest()){
-            StorageReference ref = getReference().child("images/"+ pictureRef.get());
-            ref.delete();
+    public Task<Void> deleteImageFromStorage(ObservableField<String> pictureRef, String category) {
+
+        if(Storage.checkStoragePath(category) && pictureRef != null && pictureRef.get() != null){
+            String path = category.toLowerCase()+"/";
+            StorageReference ref = getReference().child(path+pictureRef.get());
+            return ref.delete();
+        }
+        return null;
+    }
+
+    @Override
+    public Bitmap getPictureFromDevice(int requestCode, int resultCode, Intent data, Context context, ImageView view) {
```

Comment:
The use of request codes is ugly here. Can't we try to use an enum? I think that it would be a bit cleaner. Using static final int's is better than hardcoding. If the values need to be the ones that you set, try to make an enum with a method that returns a private local variable.

Reply:
I'm not sure to see what you mean, the request codes are already declared as static ints if I'm not mistaken

Reply:
I agree as I commented above

Reply:
Its' an proveide can't change signature